### PR TITLE
[GO][Client] Multipart/form-data Request Support More Than One File

### PR DESCRIPTION
**Pull request by icubbon**
The Go client will iterate over the files for a multipart/form-data request and add each file to a slice of `formFiles`.
Added a private struct `formFile` that contains the already defined values of a `formFile`. A Request now creates an empty slice of `formFiles` in which files are appended into.
The `prepareRequest` function will now iterate over the slice of `formFiles` to add each one to the request.

Addresses https://github.com/OpenAPITools/openapi-generator/issues/8322

An example snippet of a schema that has the problem being addressed, note the second file in the multipart/form-data request body:
```
  "/pet/{petId}/uploadImage":
    post:
      tags:
        - pet
      summary: uploads an image
      description: ""
      operationId: uploadFile
      parameters:
        - name: petId
          in: path
          description: ID of pet to update
          required: true
          schema:
            type: integer
            format: int64
      responses:
        "200":
          description: successful operation
          content:
            application/json:
              schema:
                $ref: "#/components/schemas/ApiResponse"
      security:
        - petstore_auth:
            - "write:pets"
            - "read:pets"
      requestBody:
        content:
          multipart/form-data:
            schema:
              type: object
              properties:
                additionalMetadata:
                  description: Additional data to pass to server
                  type: string
                file:
                  description: file to upload
                  type: string
                  format: binary
                secondFile:
                  description: another file to upload
                  type: string
                  format: binary
```

@antihax @grokify @kemokemo @jirikuncar @ph4r5h4d 

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (5.3.0), `6.0.x`
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/go/api.mustache [hunk 1]
```diff
@@ -251,22 +249,28 @@ func (a *{{{classname}}}Service) {{nickname}}Execute(r {{#structPrefix}}{{&class
 {{/headerParams}}
 {{#formParams}}
 {{#isFile}}
-	localVarFormFileName = "{{baseName}}"
+    var {{paramName}}LocalVarFormFileName string
+    var {{paramName}}LocalVarFileName     string
+    var {{paramName}}LocalVarFileBytes    []byte
```
Comment (jirikuncar) [inline modules/openapi-generator/src/main/resources/go/api.mustache:254]: please use tabs in Go templates

**Review by jirikuncar — COMMENTED** [review]
Can you create a schema with an endpoint that is using multiple form files?

**Comment by icubbon** [comment]
> Can you create a schema with an endpoint that is using multiple form files?

Do you want an the PetStore example updated or do you want to see it as part of this PR description?
Thanks!

**Comment by jirikuncar** [comment]
> Do you want the PetStore example updated?

The best option would be creating new folder (e.g. `multiple-data-in-forms`) in https://github.com/OpenAPITools/openapi-generator/tree/master/samples/openapi3/client/features with a simple OA3 schema.

**Comment by icubbon** [comment]
> > Do you want the PetStore example updated?
> 
> The best option would be creating new folder (e.g. `multiple-data-in-forms`) in https://github.com/OpenAPITools/openapi-generator/tree/master/samples/openapi3/client/features with a simple OA3 schema.

I added a basic single endpoint OA3 schema with the corresponding Go Client go into `samples/openapi3/client/features/multiple-data-in-forms`. Can you take a look at let me know if that's what you were thinking?
Thanks!

File: modules/openapi-generator/src/test/resources/3_0/petstore.yaml [hunk 2]
```diff
@@ -283,6 +283,10 @@ paths:
                   description: file to upload
                   type: string
                   format: binary
+                secondFile:
+                  description: another file to upload
+                  type: string
+                  format: binary
```
Comment (jirikuncar) [inline modules/openapi-generator/src/test/resources/3_0/petstore.yaml:289]: Do we want to add it to this file? It will affect all clients.

Comment (icubbon) [reply modules/openapi-generator/src/test/resources/3_0/petstore.yaml]: I went ahead and removed it. Looking through active and closed bugs, it looks like other clients have been impacted by this issue to.

**Comment by wing328** [comment]
LGTM but I've not thoroughly tested it locally

**Pull request merged**